src/sys/epoll.rs

```diff
@@ -72,9 +87,10 @@ pub fn epoll_create1(flags: EpollCreateFlags) -> Result<RawFd> {
 }
 
 #[inline]
-pub fn epoll_ctl(epfd: RawFd, op: EpollOp, fd: RawFd, event: &mut EpollEvent) -> Result<()> {
-    let res = unsafe { libc::epoll_ctl(epfd, op as c_int, fd, &mut event.event) };
-
+pub fn epoll_ctl<'a, T>(epfd: RawFd, op: EpollOp, fd: RawFd, event: T) -> Result<()>
+    where T: Into<&'a mut EpollEvent>
+{
+    let res = unsafe { libc::epoll_ctl(epfd, op as c_int, fd, &mut event.into().event) };
```

Comment:
In order for this to be safe, we should check, if `op` is `EPOLL_CTL_DEL` when `event` is NONE. Otherwise we might pass a null pointer into `libc::epoll_ctl`, when it expects something else.